### PR TITLE
3

### PR DESCRIPTION
**Pull request by Geloger37**
Сделал все по коду, но пока не удалось достичь 1.4. Постараюсь успеть включительно до 30

### CODE REVIEW

**Comment by Geloger37** [comment]
Получилось сделать ошибку на валидации меньше 1.4

**Comment by Geloger37** [comment]
> в score забыл torch.no_grad и self.eval(). Не забывай, что перед скорингом pytorch модели **всегда** надо переводить ее в эвал и отключать градиенты. 

Спасибо за важное замечание! Получается сейчас я нес лишние градиенты и вычислительный граф, можно повысить скорость выполнения.

> В чем особенности оптимайзера Lion ? Почему ты решил применить его для дообучения ? 

Особенности у него, как минимум, в следующем:
- Требует меньше памяти, так как не хранит средние градиента и квадрата градиента
- Меньшая скорость обучения по сравнению с адаптивным оптимизатором
- Проще алгоритм, который был получен генеративным алгоритмом
- В отличие от Adam не подстраивается под размер градиентов, но использует два шага усреднения
- Хотя в статье о нем написано, что он не слишком хорошо работает с маленьким размером пакета и большим количеством аугментацией (а я аугментировал 100% обучающих данных), но я подумал по предыдущим графикам обучения с помощью Adam модель можно дообучить.
Изначально решил это сделать с AdamW, но уменьшенным lr, но эксперимент был провальным (не включен в тетрадку) и возникло несколько идей, которые нужно было проверить. Одна из - попробовать убрать аугментацию и оставить чистые данные, но была вероятность быстро переобучиться или...
Я читал, что в реальности этот оптимизатор хорошо работает с трансформерными моделями и имеет хорошую репутацию при find-tuning моделей, а также обобщающей способностью. Вот последнее и захотелось проверить, ведь у меня есть аугментация (не позволит переобучиться, но при хорошей обобщающей способности сможет уловить распределение данных) и довольно большой loss на обучающей выборке.

**Comment by medbar** [comment]
Баллы выставил :+1:
